### PR TITLE
Check html-lint API is up before running

### LINKED ISSUE #5: Check html-lint API is up before running
Do a check to ensure the mozilla html lint API is up before invoking linting.  See also #4

### PR DESCRIPTION
**Pull request by rstoenescu**
Fixes #5

### CODE REVIEW

**Comment by rstoenescu** [comment]
Done. Using `apiAvailable` variable now. Take a look.

File: lib/gulp-html5-lint.js [hunk 1]
```diff
@@ -5,14 +5,35 @@ var _ = require('ls-lodash'),
...
 module.exports = function gulpHtml5Lint(options) {
     var linterOpts = _.safeMerge(options),
-        errorOutputList = [];
+        errorOutputList = [],
+        req,
+        apiAvailable;
+
+    try {
+        req = request('GET', linterOpts.website || 'https://html5.validator.nu', {
+            timeout: 3000
```
Comment (SpainTrain) [inline lib/gulp-html5-lint.js]: It's unclear what this value should be given that html5-lint calls `request` with no `timeout` option.  Testing from different geographies, the response time is sometimes greater than 3 seconds, so we may want to up this to 10 seconds or so for folks with high latency.

Comment (SpainTrain) [inline lib/gulp-html5-lint.js]: nit (optional): I really like to be explicit and fluent with time values, e.g.:

``` js
var moment = require('moment');
....
timeout: moment.duration(10, 'seconds').asMilliseconds()
```

File: lib/gulp-html5-lint.js [hunk 2]
```diff
@@ -5,14 +5,35 @@ var _ = require('ls-lodash'),
...
+        req = {
+            statusCode: 0
+        };
+    }
+    finally {
+        apiAvailable = req.statusCode === 200;
+    }
+
+    if (!apiAvailable) {
+        gutil.log('html-lint API is DOWN. Skipping linting.');
```
Comment (SpainTrain) [inline lib/gulp-html5-lint.js]: I think we need an option that flags whether API down is an error or just a warning (as it is here).  For our workflow, I would want the CI pipeline to stop if all linters were not run, but it also makes total sense that there are many workflows where you don't want to fail when the linter cannot run.

**Comment by mike-spainhower** [comment]
Thanks for the PR!  This is awesome!  I left some comments (sorry, used my other account, hah).  Please let me know what you think and we can get this merged.

File: test/gulp-html5-lint.test.js [hunk 3]
```diff
@@ -60,4 +61,29 @@ describe('gulp-html5-lint', function() {
         stream.write(context.gutilFile);
         stream.end();
     });
+
+    it('should log a warning when dependency website is down', function(done) {
```
Comment (mike-spainhower) [inline test/gulp-html5-lint.test.js]: Thanks for adding a test :ok_hand:

**Comment by rstoenescu** [comment]
- Added option to trigger error (+1 test to cover it)
- Timeout is 10 seconds by default, but configurable

**Comment by mike-spainhower** [comment]
Looks great, thanks for making those changes!

@rstoenescu do you mind squashing into one commit?  Once done I can merge and publish!

**Comment by rstoenescu** [comment]
Done. You can merge. Thanks.

**Pull request merged**